fastssl/data/simple_dataloaders.py

```diff
@@ -0,0 +1,35 @@
+import numpy as np
+from torch.utils.data import Dataset, DataLoader
+
+class SimpleDataset(Dataset):
+    def __init__(self,
+                 data_labels_dict,
+                 X_key='activations',
+                 y_key='labels'):
+        self.X = data_labels_dict[X_key]
+        self.y = data_labels_dict[y_key]
+
+    def __len__(self):
+        return len(self.X)
+
+    def __getitem__(self,idx):
+        return self.X[idx], self.y[idx]
+    
+def SimpleDataloader(fname_train,
+                     fname_test,
+                     splits=['train','test'],
+                     batch_size=512,
+                     num_workers=2):
+    assert fname_train==fname_test, "Precaching train/test features should be stored in the same file!"
```

Comment:
should we pass a single file in this case? or do you want to support diff files in the future?

Reply:
Discussed and ignored this suggestion.